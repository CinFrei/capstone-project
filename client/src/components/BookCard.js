import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

import RectangleButton from './RectangleButton'

/** @component */
export default BookCard

BookCard.propTypes = {
  author: PropTypes.array,
  buttonMoreName: PropTypes.string,
  buttonName: PropTypes.string,
  cover: PropTypes.string,
  handleMoreButtonClick: PropTypes.func,
  onCardButtonClick: PropTypes.func,
  title: PropTypes.string,
}

function BookCard({
  author,
  buttonMoreName,
  buttonName,
  cover,
  handleMoreButtonClick,
  onCardButtonClick,
  title,
}) {
  return (
    <BookCardWrapper>
      <CoverStyled src={cover} alt={title} />
      <div>
        <h3>{title}</h3>
        <span>
          von<h4>{author}</h4>
        </span>
      </div>
      {/* DeleteButton */}
      <RectangleButton
        buttonName={buttonName}
        gridColumn="2 / -1"
        gridPosition="end"
        onClick={onCardButtonClick}
      />
      {/* DetailButton */}
      <RectangleButton
        buttonName={buttonMoreName}
        gridColumn="2 / -1"
        gridPosition="end"
        onClick={handleMoreButtonClick}
      />
    </BookCardWrapper>
  )
}

const BookCardWrapper = styled.li`
  display: grid;
  grid-gap: 10px;
  grid-template-columns: 40% 60%;
  grid-template-rows: 1fr 45px;
  min-width: 0;
  padding: 10px;
  position: relative;

  h3 {
    margin: 0;
  }

  h4 {
    display: inline;
    margin: 10px;
  }

  span {
    margin: 0 0 0 3.5px;
  }

  p {
    margin: 15px 5px 0 3px;
  }
`

const CoverStyled = styled.img`
  grid-column: 1 / span 1;
  grid-row: 1 / span 2;
  justify-self: stretch;
`
