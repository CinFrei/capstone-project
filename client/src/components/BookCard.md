BookCard Add Book:

```js
<BookCard
  author="Andrzej Sapkowski"
  buttonName="Buch hinzufÃ¼gen."
  cover="./public/styleguide/sample-cover.jpg"
  title="Das Schwert der Vorsehung"
/>
```

BookCard Delete Book:

```js
<BookCard
  author="Andrzej Sapkowski"
  buttonName="Buch entfernen."
  cover="./public/styleguide/sample-cover.jpg"
  title="Das Schwert der Vorsehung"
/>
```
