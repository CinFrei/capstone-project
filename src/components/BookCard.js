import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

import { ReactComponent as DeleteIcon } from '../assets/delete-icon.svg'
import CircleButton from './CircleButton'

/** @component */
export default BookCard

BookCard.propTypes = {
  coverUrl: PropTypes.string,
  title: PropTypes.string,
  author: PropTypes.string,
  description: PropTypes.string,
}

function BookCard({ cover, title, author, description, onDelete, id }) {
  return (
    <BookCardWrapper>
      <StyledCover src={cover} alt={title} />
      <div>
        <h3>{title}</h3>
        <span>
          von<h4>{author}</h4>
        </span>
        <p>{description}</p>
      </div>
      <CircleButton
        onClick={onDelete}
        buttonIcon={<DeleteIcon />}
        gridColumn="2 / -1"
        gridPosition="end"
      />
    </BookCardWrapper>
  )
}

const BookCardWrapper = styled.div`
  position: relative;
  min-width: 0;
  display: grid;
  grid-template-columns: 40% 60%;
  grid-template-rows: 1fr 45px;
  grid-gap: 10px;
  padding: 10px;
  background: var(--primary-white);
  background: var(--primary-background-gradient);
  border: 2px solid var(--dark-blue);
  border-radius: 3px;

  h3 {
    margin: 0;
    color: var(--dark-blue);
  }

  h4 {
    display: inline;
    margin: 10px;
  }

  span {
    margin: 0 0 0 3.5px;
  }

  p {
    margin: 15px 5px 0 3px;
  }
`

const StyledCover = styled.img`
  grid-column: 1 / span 1;
  grid-row: 1 / span 2;
  justify-self: stretch;
`
