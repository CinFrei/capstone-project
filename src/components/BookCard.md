Basic BookCard:

```js
<BookCard
    coverUrl="./public/styleguide/sample-cover.jpg"
    title="Das Schwert der Vorsehung"
    author="Andrzej Sapkowski"
    description='Untiere, Dämonen, Spannung und Witz "Was soll man von dir halten - ein Hexer, der jeden zweiten Tag ein einträgliches Angebot ausschlägt?"'
/>
```

