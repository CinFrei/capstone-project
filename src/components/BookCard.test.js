import BookCard from './BookCard'
import { render } from '@testing-library/react'

describe('BookCard', () => {
  it('renders correctly', () => {
    const { container } = render(
      <BookCard
        cover="beautiful picture"
        title={['Paul Tobin']}
        author="Buchautor"
        buttonName="Buch hinzufÃ¼gen."
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
