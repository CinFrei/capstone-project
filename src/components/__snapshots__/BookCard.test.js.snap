// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard renders correctly 1`] = `
.c0 {
  position: relative;
  min-width: 0;
  display: grid;
  grid-template-columns: 40% 60%;
  grid-gap: 10px;
  padding: 10px;
  background: var(--primary-white);
  background: var(--primary-background-gradient);
  border: 2px solid var(--dark-blue);
  border-radius: 3px;
}

.c0 h3 {
  margin: 10px;
}

.c0 h4 {
  display: inline;
  margin: 10px;
}

.c0 img {
  width: 100%;
}

<div
  class="c0"
>
  <img
    alt=""
    src="blub"
  />
  <div>
    <h3>
      Buchtitel
    </h3>
    <span>
      von
      <h4>
        Buchautor
      </h4>
    </span>
    <p>
      Buchbeschreibung
    </p>
  </div>
</div>
`;
